internal/logging/ctx.go

```diff
@@ -42,14 +44,9 @@ func WithEdgeRequestId(ctx context.Context, id string) context.Context {
 
 // TraceId returns request id or an empty string when not set.
 func TraceId(ctx context.Context) string {
-	value := ctx.Value(requestIdCtxKey)
-	if value == nil {
+	value := trace.SpanContextFromContext(ctx).TraceID()
+	if !value.IsValid() {
 		return ""
 	}
-	return value.(string)
-}
-
-// WithTraceId returns context copy with trace id value.
-func WithTraceId(ctx context.Context, id string) context.Context {
-	return context.WithValue(ctx, requestIdCtxKey, id)
+	return value.String()
 }
```

Comment:
This makes sense, also delete the key constant.

Reply:
Done :)